be/src/storage/primary_key_encoder.cpp

```diff
@@ -526,9 +526,12 @@ bool PrimaryKeyEncoder::encode_exceed_limit(const Schema& schema, const Chunk& c
     std::vector<const void*> datas(ncol, nullptr);
     if (ncol == 1) {
         if (schema.field(0)->type()->type() == TYPE_VARCHAR) {
```

Comment:
You can add a unit test case for it.

Reply:
> You can add a unit test case for it.

done, finished in https://github.com/StarRocks/starrocks/pull/24680/commits/c138702863f4871bb32ef525a4ed8f2f9333d167